it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/BusinessIdentificationControllerISpec.scala

```diff
@@ -583,6 +586,77 @@ trait BusinessIdentificationControllerISpec extends BaseISpec with SessionDataMi
 
         result.header.headers(LOCATION) shouldBe routes.BusinessIdentificationController.showUpdateBusinessAddressForm().url
       }
+
+      "redirect to showUpdateBusinessAddressForm if the user has clean creds and isSubscribedToAgentServices=false and" when {
+        "ETMP record contains blacklisted postcode" in {
+          implicit val request = authenticatedAs(subscribingAgentEnrolledForNonMTD)
+            .withSession("businessType" -> "sole_trader")
+            .withFormUrlEncodedBody("confirmBusiness" -> "yes")
+          sessionStoreService.currentSession.knownFactsResult = Some(
+            KnownFactsResult(
+              utr = Utr("0123456789"),
+              postcode = "AA11AA",
+              taxpayerName = "My Agency",
+              isSubscribedToAgentServices = false,
+              Some(businessAddress.copy(postalCode = Some(blacklistedPostcode))),
+              None
+            ))
+
+          val result = await(controller.submitConfirmBusinessForm(request))
+
+          sessionStoreService.currentSession.initialDetails should not be empty
+
+          result.header.headers(LOCATION) shouldBe routes.BusinessIdentificationController
+            .showUpdateBusinessAddressForm()
```

Comment:
shouldnt this be routes.BusinessIdentificationController.showPostcodeNotAllowed()

Reply:
I think we need to redirect to showUpdateBusinessAddressForm from submitConfirmBusinessForm if postcode is blacklisted according to acceptance criteria